internal/resource/registry_test.go

```diff
@@ -123,3 +115,20 @@ func TestResolve(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, registration.Type, serviceType)
 }
+
+func assertRegisterPanics(t *testing.T, registerFn func(reg resource.Registration), registration resource.Registration, panicString string) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("expected panic, but none occurred")
+		} else {
+			errstr, ok := r.(string)
+			if !ok {
+				t.Errorf("unexpected error type returned from panic")
+			} else if errstr != panicString {
+				t.Errorf("expected %s error message but got: %s", panicString, errstr)
+			}
+		}
+	}()
+
```

Comment:
I think this could be replaced with [`require.PanicsWithValue`](https://pkg.go.dev/github.com/stretchr/testify/require#PanicsWithValue).

Reply:
👍 Fixed in https://github.com/hashicorp/consul/pull/17108/commits/1c66e1f3551d8b11a6e9c3a82cf5ebad60bdea23